File: androidutil/src/main/java/com/akingyin/base/utils/RandomUtil.java
package com.akingyin.base.utils;

import java.util.Random;
import java.util.UUID;

/**
 * @author king
 * @version V1.0
 * @ Description:
 * @ Date 2019/12/9 13:52
 */
public class RandomUtil {

  public static String getRandomUUID() {
    return UUID.randomUUID().toString().replace("-","").toUpperCase();
  }

  /**
   * 生成随机字符串
   *
   * @param length 生成字符串的长度
   * @return
   */
  public static String getRandomString(int length) {
    String base = "abcdefghijklmnopqrstuvwxyz0123456789";
    Random random = new Random();
    StringBuffer sb = new StringBuffer();
    for (int i = 0; i < length; i++) {
      int number = random.nextInt(base.length());
      sb.append(base.charAt(number));
    }
    return sb.toString();
  }

  /**
   * 获取随机整数
   *
   * @param min 最小值
   * @param max 最大值
   * @return
   */
  public static int getRandomNum(int min, int max) {
    return new Random().nextInt(max - min + 1) + min;
  }
}

File: androidutil/src/main/java/com/akingyin/base/utils/RandomUtil.kt
package com.akingyin.base.utils

import java.util.*

/**
 * @author king
 * @version V1.0
 * @ Description:
 * @ Date 2019/12/9 13:52
 */
object RandomUtil {
    val randomUUID: String
        get() = UUID.randomUUID().toString().replace("-", "").toUpperCase(Locale.ROOT)

    /**
    * 生成随机字符串
     *
     * @param length 生成字符串的长度
     * @return
     */
    @JvmStatic
    fun getRandomString(length: Int): String {
        val base = "abcdefghijklmnopqrstuvwxyz0123456789"
        val random = Random()
        val sb = StringBuffer()
        for (i in 0 until length) {
            val number = random.nextInt(base.length)
            sb.append(base[number])
        }
        return sb.toString()
    }

    /**
     * 获取随机整数
     *
     * @param min 最小值
     * @param max 最大值
     * @return
     */
    fun getRandomNum(min: Int, max: Int): Int {
        return Random().nextInt(max - min + 1) + min
    }
}